pdata/ptrace/traces.go

```diff
@@ -24,16 +24,23 @@ import (
 type Traces internal.Traces
 
 func newTraces(orig *otlpcollectortrace.ExportTraceServiceRequest) Traces {
-	return Traces(internal.NewTraces(orig))
+	state := internal.StateExclusive
+	return Traces(internal.NewTraces(orig, &state))
 }
 
 func (ms Traces) getOrig() *otlpcollectortrace.ExportTraceServiceRequest {
-	return internal.GetOrigTraces(internal.Traces(ms))
+	return internal.GetTracesOrig(internal.Traces(ms))
 }
 
 // NewTraces creates a new Traces struct.
 func NewTraces() Traces {
-	return newTraces(&otlpcollectortrace.ExportTraceServiceRequest{})
+	orig := &otlpcollectortrace.ExportTraceServiceRequest{}
+	state := internal.StateExclusive
+	return Traces(internal.NewTraces(orig, &state))
+}
+
+func (ms Traces) MarkShared() {
```

Comment:
MarkImmutable? or AsImmutable

Reply:
I think using `Immutable` word can be confusing here. I changed it to `AsShared` and added a comment. Let me know WDYT